lib/src/presentation/bloc/fetch_genres_bloc_impl.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:async';
+
+import '../../core/bloc/i_fetch_genres_bloc.dart';
+import '../../core/util/data_state.dart';
+import '../../core/util/enum_classes.dart';
+import '../../data/model/genres_model.dart';
+import '../../domain/entity/genre_event.dart';
+import '../../domain/usecase/usecase_impl/get_genres_usecase_impl.dart';
+
+class FetchGenreBloc extends IFetchGenresBloc {
+  final StreamController<GenresEvent> _genresFetcher =
```

Comment:
To be more clear with the naming convension instead to genresFetcher you can use like genresController